core/src/main/scala/stryker4s/extension/mutationtype/PatternMatch.scala

```diff
@@ -0,0 +1,24 @@
+package stryker4s.extension.mutationtype
+import scala.meta.Term
+
+case object PatternMatch {
+
+  def unapply(tree: Term.Match): Option[(Term, Seq[PatternMatchMutation])] = {
+    // We don't mutate a match that only has one case
+    if (tree.cases.length > 1) {
+      Some(
+        tree,
+        tree.cases.map { caze =>
+          import scala.meta.quasiquotes.XtensionQuasiquoteTerm
+
+          // https://www.scala-lang.org/api/2.12.1/scala/unchecked.html
+          // @unchecked is required because we are removing cases here,
+          // possibly resulting in compile warnings (which could be flagged as errors)
+          val annotationWrap = q"${tree.expr}: @unchecked"
```

Comment:
I really don't like that we are just flagging this as unchecked. 

When sealed traits, for example, the compiler error is already a guard against that mutator. Cases can't be accidentally removed anymore in that case. We could argue the statement in the pattern match (the executing code) is then also removed, that is very true and that should be tested. But that is move removing the actual block statement then removing the case as we do with here.

So in a case of a sealed trait and the code doesn't compile I think Stryker4s should be able to handle that and see that as a killed mutator since Scala is so strongly typed. I know this deviates from the normal stryker implementation but i could see that being different for Stryker4s what do you think @hugo-vrijswijk  @nicojs.